src/PeliasSearchManager.swift

```diff
@@ -125,18 +125,37 @@ public class PeliasResponse: APIResponse {
   let response: NSURLResponse?
   let error: NSError?
   var parsedResponse: PeliasSearchResponse?
+  var parsedError: PeliasError?
   
   init(data: NSData?, response: NSURLResponse?, error: NSError?) {
     self.data = data
     self.response = response
     self.error = error
     if let dictResponse = parseData(data) {
+      if let peliasError = parseErrorFromJSON(dictResponse) {
+        self.parsedError = peliasError
+        return
+      }
       parsedResponse = PeliasSearchResponse(parsedResponse: dictResponse)
     }
   }
   
+  private func parseErrorFromJSON(json: NSDictionary) -> PeliasError? {
+    //First get the error code
+    guard let meta = json["meta"] as? NSDictionary else { return nil }
```

Comment:
I debated on that for a bit honestly before implementing it just with literals. My thought was it's all self contained in these parsing routines and specific to the error parsing. If you feel strongly about it in the other direction though (or @sarahlensing has any thoughts on the matter) I'm open to them.